.github/workflows/build.yml

```diff
@@ -1,7 +1,27 @@
 
 name: Build
 
-on: [push, pull_request]
+on:
+  push:
+    branches:
+      - '*'
+    paths:
+      - '.github/**'
+      - 'CMakeLists.txt'
+      - 'ci/**'
+      - 'cmake/**'
+      - 'openvdb/**'
+      - 'openvdb_houdini/**'
+  pull_request:
+    branches:
+      - '*'
+    paths:
+      - '.github/**'
+      - 'CMakeLists.txt'
+      - 'ci/**'
+      - 'cmake/**'
+      - 'openvdb/**'
+      - 'openvdb_houdini/**'
```

Comment:
it would probably be better to use something like `paths-ignore: - '**.rst'  - '**.md`?

Reply:
That's where I started out, but there are exceptions that would still force a build (openvdb/CHANGES, openvdb/doc/doxygen-config), so I figured it was better to do it like this.

Reply:
CHANGES should probably be a .md file and likewise we could add a .txt extension to the dox file? Probably be a nice solution that the directory masking - better to build very frequently when unnecessary than not at all

Reply:
I'm going to just merge this for now so that we have the workflow in place, otherwise I'm playing catch up trying to keep this PR in sync with anything that gets merged in the meantime.

I don't disagree that we can do some additional reorganisation and tidy up these paths, but let's just do that in a subsequent pass if that's ok...